vip-helpers/class-wpcomvip-restrictions.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Automattic\VIP\Utils;
+
+class WPComVIP_Restrictions {
+	protected static $instance = null;
+
+	public static function instance(): self {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	private function __construct() {
+		add_filter( 'wp_insert_post_data', [ $this, 'restrict_post_author' ], PHP_INT_MAX );
+		add_filter( 'wp_insert_attachment_data', [ $this, 'restrict_post_author' ], PHP_INT_MAX );
+		add_filter( 'wp_dropdown_users_args', [ $this, 'restrict_post_author_dropdown' ], PHP_INT_MAX );
```

Comment:
The scope of `wp_dropdown_users_args` is more than just the author dropdown. I'm a little concerned it might mess with some jetpack logic wherein the `connection owner` is meant to be chosen by default for certain things. But our query filter will result in the connection owner never showing up.

We could target the post author dropdown specifically by checked the other args. Looks like `name = post_author_override` is unique.

Reply:
Noted, thanks, I will take a look on Monday.